src/deadline_worker_agent/startup/bootstrap.py

```diff
@@ -263,22 +268,63 @@ def _load_or_create_worker(
     Raises:
         SystemExit - Any error here is unrecoverable, and the Agent should exit.
     """
+    instance_id = _get_instance_id()
 
     worker_info: Optional[WorkerPersistenceInfo] = None
     has_existing_worker = False
     if use_existing_worker:
         worker_info = WorkerPersistenceInfo.load(config=config)
         if worker_info:
-            has_existing_worker = True
-            _logger.info(
-                WorkerLogEvent(
-                    op=WorkerLogEventOp.LOAD,
-                    farm_id=config.farm_id,
-                    fleet_id=config.fleet_id,
-                    worker_id=worker_info.worker_id,
-                    message="Worker identity loaded from prior run.",
+            if (
+                instance_id is not None
+                and worker_info.instance_id is not None
+                and worker_info.instance_id != instance_id
+            ):
+                # This can happen if an AMI is created from an instance that started the worker agent and left behind a
+                # worker persistence file. Ideally the file is removed before creating the AMI but in the event it still
+                # exists, simply create a new worker and replace the file.
+                _logger.info(
+                    WorkerLogEvent(
+                        op=WorkerLogEventOp.LOAD,
+                        farm_id=config.farm_id,
+                        fleet_id=config.fleet_id,
+                        message=(
+                            f"Worker state file contains an instance ID ({worker_info.instance_id}) different to that of the running host ({instance_id}). "
+                            "This could be the result of a worker state file being included in the image being launched. "
+                            "For guidance on preparing a worker image, please refer to https://docs.aws.amazon.com/deadline-cloud/latest/developerguide/create-ami.html#prepare-the-instance. "
+                            "Ignoring the persisted worker identity and creating a new worker. "
+                        ),
+                    )
```

Comment:
Two things:

1.  I think we should include the problematic `worker_id` here, for structured logging use-cases
2. This should be logged at warning level to surface this to reader's attention

What do you think?